Gopkg.toml

```diff
@@ -22,11 +22,11 @@ required = [
 
 [[constraint]]
   name="sigs.k8s.io/controller-runtime"
-  version="v0.1.1"
+  version="v0.1.3"
```

Comment:
This constraint was plain wrong, `dep` does find a solution but that is not buildable because:

```
# sigs.k8s.io/cluster-api/pkg/controller/cluster [sigs.k8s.io/cluster-api/pkg/controller/cluster.test]
pkg/controller/cluster/controller.go:129:43: unknown field 'RequeueAfter' in struct literal of type reconcile.Result
# sigs.k8s.io/cluster-api/pkg/controller/machine [sigs.k8s.io/cluster-api/pkg/controller/machine.test]
pkg/controller/machine/controller.go:168:44: unknown field 'RequeueAfter' in struct literal of type reconcile.Result
# sigs.k8s.io/cluster-api/cmd/clusterctl/cmd
cmd/clusterctl/cmd/validate_cluster.go:72:91: mgr.GetRESTMapper undefined (type manager.Manager has no field or method GetRESTMapper)
# sigs.k8s.io/cluster-api/cmd/clusterctl/validation [sigs.k8s.io/cluster-api/cmd/clusterctl/validation.test]
cmd/clusterctl/validation/validate_cluster_api_objects_test.go:424:79: mgr.GetRESTMapper undefined (type manager.Manager has no field or method GetRESTMapper)
# sigs.k8s.io/cluster-api/cmd/clusterctl/cmd [sigs.k8s.io/cluster-api/cmd/clusterctl/cmd.test]
cmd/clusterctl/cmd/validate_cluster.go:72:91: mgr.GetRESTMapper undefined (type manager.Manager has no field or method GetRESTMapper)
```

The first version of `kubebuilder` that does have a field `RequeueAfter` in the `reconcile.Result` struct is `v0.1.3`: https://github.com/kubernetes-sigs/controller-runtime/commit/83f4800636a002e27cc45ae72406285d570da1b1

Reply:
Why is this constraint wrong? It's in the section with `STANZAS BELOW ARE GENERATED AND MAY BE WRITTEN - DO NOT MODIFY BELOW THIS LINE.` which is auto-generated by kubebuilder.

@pwittrock @droot

Reply:
This is an upstream issue, the versions in the generated `Gopkg.toml` are hardcoded: kubernetes-sigs/kubebuilder/issues/419

However, we can just remove the `# STANZAS BELOW ARE GENERATED AND MAY BE WRITTEN - DO NOT MODIFY BELOW THIS LINE.` line from the `Gopkg.toml` at least until the upstream issue gets resolved. When doing that, running `kubebuilder update` results in a

```
$ kubebuilder update
2018/10/07 11:16:50 error updating vendor dependecies skipping modifying Gopkg.toml - file already exists and is unmanaged
```

Any objections to that?

Reply:
Explained the reasoning why the version says `v0.1.1` in the Stanza in the issue here https://github.com/kubernetes-sigs/kubebuilder/issues/419#issuecomment-427922785

In order to bring deps in sync, `dep ensure -update sigs.k8s.io/controller-runtime sigs.k8s.io/controller-tools` should be enough to fetch latest version.